.github/workflows/build-images-workflow-run.yml

```
@@ -209,6 +209,18 @@ jobs:
         uses: actions/checkout@v2
         with:
           ref: ${{ needs.cancel-workflow-runs.outputs.sourceHeadSha }}
+      - name: "Retrieve DEFAULTS from the _initialization.sh"
+        # We cannot "source" the script here because that would be a security problem (we cannot run
+        # any code that comes from the sources coming from the PR. Therefore we extract the
+        # DEFAULT_BRANCH and DEFAULT_CONSTRAINTS_BRANCH via custom grep/awk/sed commands
+        run: |
+          DEFAULT_BRANCH=$(grep "export DEFAULT_BRANCH" scripts/ci/libraries/_initialization.sh | \
+            awk 'BEGIN{FS="="} {print $3}' | sed s'/["}]//g')
```

Could be done as a single awk command:


```suggestion
          DEFAULT_BRANCH=$(awk -F= '/export DEFAULT_BRANCH/ {gsub(/"/, "", $2); print $2}' \
              scripts/ci/libraries/_initialization.sh)

```

---

Not so sure :) (See above). But  I am happy to play a game in a) the shortest  b) the least understandable c) the most explicit way of doing it :)

---

Pet peave of mine is the `grep | awk` given that awk has the ability build in. Not critical of course.

---

Yeah, You know I am  of the old good POSIX/unix philosophy - combine many small tools that do one thing very well :).